File: Trabalho_Faculdade/questao2.java
class Questao2 {
  public static void main(String[] args) {
      int[] numeros = { 1000000, 10000000, 100000000, 1000000000 };

      for (int valor : numeros) {
          long inicio_tempo = System.currentTimeMillis();
          long soma = 0;

          for (long i = 1; i <= valor; i++) {
              soma += i;
          }

          long tempo_final = System.currentTimeMillis();
            
          long tempo_execucao = tempo_final - inicio_tempo;

          System.out.println("Soma dos números de 1 até " + valor + ": " + soma);
          System.out.println("Tempo de execução: " + tempo_execucao + " ms");
          System.out.println();
      }
  }
}

File: Trabalho_Faculdade/questao_2.py
import time

def somar_numeros(valor):
    tempo_inicio = time.time()
    soma = sum(range(1, valor + 1))
    tempo_final = time.time()
    tempo_execucao = tempo_final - tempo_inicio
    return soma, tempo_execucao

valores = [1000000, 10000000, 100000000, 1000000000]

for valor in valores:
    soma_result, tempo_execucao = somar_numeros(valor)
    print(f"Soma dos números de 1 até {valor}: {soma_result}")
    print(f"Tempo de execução: {tempo_execucao} segundos")
    print()
